src/lib/netlist/plib/ppmf.h

```diff
@@ -106,45 +64,53 @@ namespace plib {
 	{
 		using ci = compile_info;
 		enum { value =
+			(PPMF_FORCE_TYPE >= 0)                                               ? PPMF_FORCE_TYPE :
 			(ci::type() == ci_compiler::CLANG && !ci::m64()
-				&& ci::os() == ci_os::WINDOWS)                                   ? 0 :
-			(ci::mingw() && !ci::m64() && ci::version() >= 407)                  ? 0 :
-			(ci::mingw() && !ci::m64())                                          ? 1 :
+				&& ci::os() == ci_os::WINDOWS)                                   ? PPMF_TYPE_PMF :
+			(ci::mingw() && !ci::m64() && ci::version() >= 407)                  ? PPMF_TYPE_PMF :
+			(ci::mingw() && !ci::m64())                                          ? PPMF_TYPE_PMF : // Dropped support for mingw32 < 407 PPMF_TYPE_INTERNAL_ITANIUM :
 			((ci::type() == ci_compiler::CLANG || ci::type() == ci_compiler::GCC)
 				&& (ci::arch() == ci_arch::MIPS
 					|| ci::arch() == ci_arch::ARM
-					|| ci::os() == ci_os::EMSCRIPTEN))                           ? 2 :
-			(ci::type() == ci_compiler::CLANG || ci::type() == ci_compiler::GCC) ? 1 :
-			(ci::type() == ci_compiler::MSC && ci::m64()) ?                        3 :
-																				   0
+					|| ci::os() == ci_os::EMSCRIPTEN))                           ? PPMF_TYPE_INTERNAL_ARM :
+			(ci::type() == ci_compiler::CLANG || ci::type() == ci_compiler::GCC) ? PPMF_TYPE_INTERNAL_ITANIUM :
```

Comment:
Just noticed this - are you checking for clang compiling for MSVC ABI (e.g. using `clang-cl` wrapper)? In that situation, you need to use MSVC ABI but still avoid casting pointers to member functions across virtual inheritance relationships as clang doesn't support that MSVC extension.

Reply:
Could you provide an example of which defines are set by clang-cl to identify the build?

Reply:
Both `_MSC_VER` and `__clang__` are defined in that situation.  For example the situation is tested for to avoid attempting a test that only works with MSVC itself here: https://github.com/mamedev/mame/blob/mame0243/src/emu/validity.cpp#L1415

(This configuration is fairly popular with people who like the Visual Studio debugger, but want to avoid the long compile times and poor code generation of MSVC.)

Reply:
Supported now (see https://github.com/mamedev/mame/pull/9752/commits/49e7da33f600b61d84292bf66245ccccc068b2ad)